sycl/doc/extensions/experimental/sycl_ext_oneapi_group_load_store.asciidoc

```diff
@@ -442,6 +442,23 @@ so the implementation can rely on `get_max_local_range()` range size:
 
 If partition is uneven the behavior is undefined.
 
+== Alignment
+
+The following property can be used to provide an alignment of the pointer.
+It allows to avoid dynamic alignment check.
```

Comment:
The APIs in this extension take an iterator, which is not necessarily a pointer.  I think the `alignment` property only makes sense when the iterator is a pointer.  What happens if it is not?  It seems like you should be able to detect this at compile time.  Will it raise a static assertion?  If so, we should add a Mandates paragraph to each function like:

> _Mandates:_ If `Properties` contains the `alignment` property, `OutputIteratorT` must be a pointer.

Alternatively, we could add one more constraint like:

> * If `OutputIteratorT` is not a pointer then `Properties` does not contain the `alignment` property.

Tagging @Pennycook to see if he has a preference between these two options.

Reply:
Most of the functions currently contain a sentence like:

> Properties may also provide assertions which can enable better optimizations.

Should we expand this to:

> Properties may also provide assertions or the `alignment` property which can enable better optimizations.

Reply:
Indeed, thank you for pointing that out!
Updated according to your suggestion. I've chosen the first option with Mandates and static assert in the code.